logic/common/log_utils.py

```diff
@@ -0,0 +1,188 @@
+''' Utility functions for logging purposes
+
+    <To Import>
+import logic.common.utils_log as log
+
+'''
+
+import logging
+
+#--------------------------------------------------#
+'''Logger Settings'''
+
+FOLDER_LOG = 'reference/log/'
+
+DEBUG_1 = False    # Prints messages in SetVerbosityLevel() 
+
+
+
+#--------------------------------------------------#
+'''Configuration'''
+
+def SetVerbosityLevel(log_lv_num, file_name = 'logfile.log', format = '%(message)s'):
+    '''Set basic configuration to logger'''
+    log_level = logging.WARNING # Default to Warning
+    if log_lv_num == 1: log_level = logging.INFO
+    elif log_lv_num == 2: log_level = logging.DEBUG
+
+    # Create a custom logger
+    logger = logging.getLogger()
+    logger.setLevel(log_level) # Set the minimum logging level
+
+    # Create a file handler to log messages to a file
+#    file_handler = logging.FileHandler(FOLDER_LOG+file_name)
+    file_handler = logging.FileHandler(FOLDER_LOG+file_name, 'w')
+    file_handler.setLevel(log_level)
+    file_handler.setFormatter(logging.Formatter(format))
+
+    # Create a stream handler for console output
+    console_handler = logging.StreamHandler()
+    console_handler.setLevel(log_level)
+    console_handler.setFormatter(logging.Formatter(format))
+
+    # Add the handlers to the logger
+    logger.addHandler(file_handler)
+    logger.addHandler(console_handler)
+
+    # Debug
+    _LoggingConfigMessage(log_lv_num)
+    if not DEBUG_1: return
+    Must("Message 0")
+    Info("Message 1")
+    Extra("Message 2")
+
+def _LoggingConfigMessage(log_lv_num):
+    print_msg = ""
+    print_msg += "  "                             # Indentation
+    print_msg += "Verbosity = " + str(log_lv_num) # Verbosity Level
+    print_msg += "" # TODO DateTime
+    logging.warning(print_msg)
+
+
+
+
+#--------------------------------------------------#
+'''Public Functions'''
+
+def Must(s):
+    '''activated no matter what'''
+    print_str = ConvertToLoggingString(s)
+    logging.warning(print_str)
+
+
+def Info(s):
+    '''activated when --v = 1 or 2 (normal level of verbosity)'''
+    print_str = ConvertToLoggingString(s)
+    logging.info(print_str)
+
+
+def Extra(s):
+    '''activated only when -v = 2 (extra verbose)'''
+    print_str = ConvertToLoggingString(s)
+    logging.debug(print_str)
+
+
+
+#--------------------------------------------------#
+'''Utility'''
+
+def ConvertToLoggingString(s):
```

Comment:
I'm going to reject this function (see Trello for the meta why). But for educational purposes, I'll review its nitty-gritty parts too, so see my comments below